client/state/partner-portal/partner/hooks/use-update-company-details.ts

```diff
@@ -0,0 +1,37 @@
+import { useMutation, UseMutationOptions, UseMutationResult } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import { APIError, APIPartner } from 'calypso/state/partner-portal/types';
+
+interface MutationUpdateCompanyDetailsVariables {
+	name?: string;
+	country?: string;
+	city?: string;
+	line1?: string;
+	line2?: string;
+	postal_code?: string | number;
```

Comment:
The way I think about this is based on what is internal and what is external.
Anything in Calypso should be in Calypso's format. Anything coming in from the "outside" or going out should be in whatever format the "outside" in that case uses.

As such, JSON coming in from the WPCOM API is in snake_case and when it enters Calypso's domain, it should be converted to an appropriate format for Calypso.
When data needs to go out to the WPCOM API it should be in snake_case as that is what the API uses, even if that means we have to convert it back to snake_case.
The same logic applies to Tracks - it's an "outside" domain and we need to convert it if that is what that domain expects.

For argument's sake, let's say that Stripe used kebab-case and Tracks used MACRO_CASE - which one would we use? Would we allow our Calypso state to be littered with 4 different cases because of where that data originally comes from? Even if that data comes from someplace else the moment it enters Calypso's domain, it becomes Calypso's data and should be treated as such.

Isolating and converting data's structure at the entry and exit points makes the most sense to me, even if that means some extra work on converting the data's structure on the way in and on the way out.